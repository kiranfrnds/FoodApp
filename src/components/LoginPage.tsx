import React, { useState } from "react";
import PhoneIcon from "@mui/icons-material/Phone";
import { useNavigate } from "react-router";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

function LoginPage() {
  const [number, setNumber] = useState<any>("");

  const navigate = useNavigate();

  const onClickHandle = async (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    let userData = await fetch(
      "https://extended-retail-app.herokuapp.com/api/customer/getOtp",
      {
        method: "POST",
        body: JSON.stringify({ username: number }),
        headers: {
          "Content-Type": "application/json",
          Accept: "application/json",
        },
      }
    );
    userData = await userData.json();
    if (userData.status === 200) {
      localStorage.setItem("mobileNumber", number);
      toast("OTP sent to your mobile number");
      navigate({
        pathname: `/verification/${number}`,
      });
    } else {
      toast("Please Enter valid Number ‚ùå");
    }
  };

  return (
    <div className="bg-container">
      <div className="bottom-bg">
        <img
          src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652702958/logo_1_seaocx.png"
          alt="food-app"
          className="food-logo"
        />
        <h4 className="main-heading">Welcome Back</h4>
        <h5 className="login-desc">Login Account</h5>
        <div className="login-items">
          <div className="login">
            <img
              src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652702965/1200px-Flag_of_India_1_dfhmj5.png"
              alt="india"
              className="india-log"
            />
            <p className="num-logo">+91 </p>
            <img
              src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652708990/Vector_5_vvb70x.png"
              alt="fold"
              className="fold"
            />
            <hr className="line" />
          </div>

          <div className="items">
            <input
              type="number"
              placeholder="Mobile No"
              className="input"
              value={number}
              onChange={(e) => setNumber(e.target.value)}
            />
            <PhoneIcon className="phone-icon" />
          </div>
        </div>
        <button
          onClick={onClickHandle}
          disabled={number.length < 10}
          className={number.length < 10 ? "disable" : "enable"}
        >
          GET OTP
        </button>
      </div>
      <div className="bg-image">
        <img
          src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652709778/Online_Shoping_29-ai_1_ijnxly.png"
          alt="order"
          className="order-icon"
        />
      </div>
      <ToastContainer />
      <div></div>
    </div>
  );
}

export default LoginPage;
