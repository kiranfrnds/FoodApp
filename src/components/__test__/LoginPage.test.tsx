import { fireEvent, render, screen } from "@testing-library/react";
import LoginPage from "../LoginPage";
import { BrowserRouter as Router } from "react-router-dom";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

async function GetOtp(number: any) {
  try {
    const result = await fetch(
      `https://extended-retail-app.herokuapp.com/api/customer/getOtp`,
      {
        method: "POST",
        body: JSON.stringify({ username: number }),
        headers: {
          "Content-Type": "application/json",
          Accept: "application/json",
        },
      }
    );
    const data = await result.json();
    return data.status === 200;
  } catch (err) {
    return null;
  }
}

beforeAll(() => {
  jest.spyOn(window, "alert").mockImplementationOnce(() => {});
  jest.spyOn(console, "error").mockImplementationOnce(() => {});
});

test("snapshot", () => {
  const { asFragment } = render(
    <Router>
      <LoginPage />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders login page", () => {
  render(
    <Router>
      <LoginPage />
    </Router>
  );
  expect(screen.getByText("Login Account")).toBeInTheDocument();
});

test("renders input field", () => {
  render(
    <Router>
      <LoginPage />
    </Router>
  );
  expect(screen.getByPlaceholderText("Mobile No")).toBeInTheDocument();
});

test("after entering mobile number, renders getotp button", () => {
  render(
    <Router>
      <LoginPage />
    </Router>
  );
  const input = screen.getByPlaceholderText("Mobile No");
  const button = screen.getByRole("button");
  fireEvent.change(input, { target: { value: "1234567891" } });
  expect(button).toBeEnabled();
});

test("after clicking button goes to verification page", async () => {
  render(
    <Router>
      <LoginPage />
    </Router>
  );
  const input = screen.getByPlaceholderText("Mobile No");
  const button = screen.getByRole("button");
  fireEvent.change(input, { target: { value: "1234567891" } });
  fireEvent.click(button);
  const result = await GetOtp("1234567891");
  expect(result).toBeTruthy();
});
