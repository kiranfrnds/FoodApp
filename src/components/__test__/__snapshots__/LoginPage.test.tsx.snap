// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-container"
  >
    <div
      class="bottom-bg"
    >
      <img
        alt="food-app"
        class="food-logo"
        src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652702958/logo_1_seaocx.png"
      />
      <h4
        class="main-heading"
      >
        Welcome Back
      </h4>
      <h5
        class="login-desc"
      >
        Login Account
      </h5>
      <div
        class="login-items"
      >
        <div
          class="login"
        >
          <img
            alt="india"
            class="india-log"
            src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652702965/1200px-Flag_of_India_1_dfhmj5.png"
          />
          <p
            class="num-logo"
          >
            +91 
          </p>
          <img
            alt="fold"
            class="fold"
            src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652708990/Vector_5_vvb70x.png"
          />
          <hr
            class="line"
          />
        </div>
        <div
          class="items"
        >
          <input
            class="input"
            placeholder="Mobile No"
            type="number"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium phone-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="PhoneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
            />
          </svg>
        </div>
      </div>
      <button
        class="disable"
        disabled=""
      >
        GET OTP
      </button>
    </div>
    <div
      class="bg-image"
    >
      <img
        alt="order"
        class="order-icon"
        src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652709778/Online_Shoping_29-ai_1_ijnxly.png"
      />
    </div>
    <div
      class="Toastify"
    />
    <div />
  </div>
</DocumentFragment>
`;
